lib/src/rules/unnecessary_lambdas.dart

```diff
@@ -80,16 +88,17 @@ class _Visitor extends SimpleAstVisitor {
 
   void _visitInvocationExpression(
       InvocationExpression node, FunctionExpression nodeToLint) {
-    if (nodeToLint.parameters.length != node.argumentList.length) {
+    if (nodeToLint.parameters.parameters.length !=
+        node.argumentList.arguments.length) {
```

Comment:
I tried, but I could not, the perfect test was the one with generics that was in the issue, but I could not reproduce it in the test because the parser could not do it, I will try to create other test without generics that reproduce the bug